puppet/modules/jenkins_job_builder/files/centos.org/jobs/foreman-katello-test.yml

```diff
@@ -1,46 +1,78 @@
+---
 - job-template:
-    name: 'foreman-katello-{version}-test'
-    project-type: pipeline
+    name: foreman-katello-{version}-{distro}-{action}-test
+    project-type: 'pipeline'
     sandbox: true
+    parameters:
+      - string:
+          name: version
+          default: '{version}'
+      - string:
+          name: distro
+          default: '{distro}'
+      - string:
+          name: action
+          default: '{action}'
+      - string:
+          name: type
+          default: 'katello'
     dsl:
       !include-raw:
-        - 'pipelines/lib/foremanCentosJob.groovy{empty}'
-        - 'pipelines/lib/pipelines.groovy{empty}'
-        - 'pipelines/test/katello/{version}.groovy'
-        - '../theforeman.org/pipelines/lib/ansible.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/duffy.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}'
-
-- job-template:
-    name: 'foreman-katello-upgrade-{version}-test'
-    project-type: pipeline
-    sandbox: true
-    dsl:
-      !include-raw:
-        - 'pipelines/lib/foremanCentosJob.groovy{empty}'
-        - 'pipelines/lib/pipelines.groovy{empty}'
-        - 'pipelines/test/katello/{version}-upgrade.groovy'
-        - '../theforeman.org/pipelines/lib/ansible.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/duffy.groovy{empty}'
-        - '../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}'
+        - pipelines/lib/foremanCentosJob.groovy{empty}
+        - pipelines/lib/pipelines.groovy{empty}
+        - ../theforeman.org/pipelines/lib/duffy.groovy{empty}
+        - ../theforeman.org/pipelines/lib/ansible.groovy{empty}
+        - ../theforeman.org/pipelines/lib/foreman_infra.groovy{empty}
 
 - project:
-    name: katello
+    name: katello-release
     jobs:
-      - 'foreman-katello-{version}-test'
-      - 'foreman-katello-upgrade-{version}-test'
+      - 'foreman-katello-{version}-{distro}-{action}-test'
     empty: ''
     version:
       - '3.13'
       - '3.14'
       - '3.15'
       - '3.16'
+    distro:
```

Comment:
wouldn't it make more sense to group by distro?

so centos7 would include all releases and nightly for install and upgrade
and centos8 would be 3.16 and nightly, install only?

Reply:
or group by "install-and-upgrade" and "install-only", so we can move 3.16 to upgradeable on el8 at some point?

Reply:
I could go that route. I think it tends to just be easier to think in terms of versions of the software rather than in distributions? Unless I misunderstand your question.

Reply:
I think what bothers me is that there is no 3.16-el8 job because 3.16 is in the "release" bucket and that has no el8 defined at all. so my second comment is probably better aligned with your original idea, it would look like this:

```yaml
- project:
    name: katello-install-el7
    jobs:
      - 'foreman-katello-{version}-{distro}-{action}-test'
    empty: ''
    version:
      - '3.14'
      - '3.15'
      - '3.16'
      - 'nightly'
      - 'devel'
    distro:
      - centos7
    action:
      - install

- project:
    name: katello-install-el8
    jobs:
      - 'foreman-katello-{version}-{distro}-{action}-test'
    empty: ''
    version:
      - '3.16'
      - 'nightly'
    distro:
      - centos8
    action:
      - install

- project:
    name: katello-upgrade
    jobs:
      - 'foreman-katello-{version}-{distro}-{action}-test'
    empty: ''
    version:
      - '3.14'
      - '3.15'
      - '3.16'
      - 'nightly'
    distro:
      - centos7
    action:
      - upgrade
```

would also fold the `devel` one into the generic install-el7 target.